app/hooks/StrapiAPI.ts

```diff
@@ -1,32 +1,280 @@
-      Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
-    });
+    fetcher<StrapiApi.StrapiResponse<T>>(
+      `${config.STRAPI_API_URL}/api/${path}`,
+      {
+        Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
+      }
+    );
 
-  const { data, error, isLoading } = useSWR<StrapiResponse<T>>(
+  const { data, error, isLoading } = useSWR<StrapiApi.StrapiResponse<T>>(
     `/api/${path}`,
     dataFetcher
   );
 
   return {
-    data: data?.data ? data.data.attributes : null,
+    data: data?.data ? data.data : null,
     error,
     isLoading,
   };
 }
 
 export function useFooterData() {
-  return useStrapiHook<Footer>("footer?populate[links][populate]=*");
+  return useStrapiHook<StrapiApi.FooterResponse>(
+    "footer?populate[links][populate]=*"
+  );
+}
+
+export function useHomepageData() {
+  return useStrapiHook<StrapiApi.HomepageResponse>(
+    "home-page?populate[hero][populate]=*&populate[category_section][populate]=*&populate[opportunity_section][populate]=*&populate[opportunities][populate][image][populate]=*&populate[opportunities][populate][address]=*&populate[opportunities][populate][links][populate]=*&populate[opportunities][populate][opportunity_categories]=*&populate[event_section][populate]=*&populate[events][populate][date]=*&populate[events][populate][address]=*&populate[events][populate][links]=*&populate[events][populate][image][populate]=*&populate[events][populate][event_categories]=*&populate[two_column_content_blocks][populate][link]=*&populate[two_column_content_blocks][populate][media][populate]=*"
+  );
+}
+
+export function usePageContent(title: string) {
+  return useStrapiHook<StrapiApi.ContentPageResponse>(
+    `content-pages?filters[title][$eq]=${title}&populate[two_column_content_blocks][populate][link]=*&populate[two_column_content_blocks][populate][media][populate]=*`
+  );
+}
+
+export namespace StrapiApi {
+  interface Meta {
+    pagination?: {
+      page: number;
+      pageSize: number;
+      pageCount: number;
+      total: number;
+    };
+  }
+
+  // Single data item response
+  export interface StrapiDatumResponse<T> {
+    id: number;
+    attributes: T;
+  }
+
+  // Data response can be a single item, an array of items, or null
+  export type StrapiDataResponse<T> =
+    | StrapiDatumResponse<T>
+    | Array<StrapiDatumResponse<T>>
+    | null;
+
+  // Unified response interface for object or array responses
+  export interface StrapiResponse<T> {
+    data: StrapiDataResponse<T>;
+    meta?: Meta;
+  }
+
+  // Specific response interface for when data is a single object (optional, but useful)
+  export interface StrapiObjectResponse<T> {
+    data: StrapiDatumResponse<T> | null;
+    meta?: Meta;
+  }
+
+  // Specific response interface for when data is an array (optional, but useful)
+  export interface StrapiArrayResponse<T> {
+    data: Array<StrapiDatumResponse<T>> | null;
+    meta?: Meta;
+  }
+
+  interface BaseAttributesResponse {
```

Comment:
This is a good observation and suggestion. Single types and items within a collection use this, so I think `BaseDatumAttributesResponse` still makes sense because it isn't single vs collection types.